src/HookRegistry.php

```diff
@@ -32,8 +32,9 @@ public function __construct( $config ) {
 	 * @since  1.0
 	 */
 	public function register() {
+		$hookContainer = MediaWikiServices::getInstance()->getHookContainer();
 		foreach ( $this->handlers as $name => $callback ) {
-			Hooks::register( $name, $callback );
+			$hookContainer->register( $name, [ $this, $callback ]);
```

Comment:
Why not

```php
$hookContainer->register( $name, $callback );
```

Reply:
I pushed an other commit

Reply:
Did you test this code?

Reply:
hello yes locally please take both 2 commits